metals/src/main/scala/scala/meta/internal/metals/QuickFix.scala

```diff
@@ -0,0 +1,66 @@
+package scala.meta.internal.metals
+
+import scala.concurrent.Future
+import scala.meta.pc.CancelToken
+import org.eclipse.{lsp4j => l}
+import scala.concurrent.ExecutionContext
+import scala.meta.internal.metals.MetalsEnrichments._
+
+trait QuickFix {
+  def contribute(
+      params: l.CodeActionParams,
+      compilers: Compilers,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]]
+}
+
+object QuickFix {
+
+  object ImportMissingSymbol extends QuickFix {
+
+    override def contribute(
+        params: l.CodeActionParams,
+        compilers: Compilers,
+        token: CancelToken
+    )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
+
+      def importMissingSymbol(
+          diagnostic: l.Diagnostic,
+          name: String
+      ): Future[Seq[l.CodeAction]] = {
+        val textDocumentPositionParams = new l.TextDocumentPositionParams(
+          params.getTextDocument(),
+          diagnostic.getRange.getEnd()
+        )
+        compilers
+          .autoImports(textDocumentPositionParams, name, token)
+          .map { imports =>
+            imports.asScala.map { i =>
+              val uri = params.getTextDocument().getUri()
+              val edit = new l.WorkspaceEdit(Map(uri -> i.edits).asJava)
+
+              val codeAction = new l.CodeAction()
+
+              codeAction.setTitle(
+                s"Import '$name' from package '${i.packageName}'"
+              )
+              codeAction.setKind(l.CodeActionKind.QuickFix)
+              codeAction.setDiagnostics(List(diagnostic).asJava)
+              codeAction.setEdit(edit)
+
+              codeAction
+            }
+          }
+      }
+
+      Future
+        .sequence(params.getContext().getDiagnostics().asScala.collect {
+          case d @ ScalacDiagnostic.SymbolNotFound(name) =>
```

Comment:
Can we quickly return if the diagnostic range does not contain the current position?

Reply:
`encloses` should be easy to use here.

Reply:
done, take a look!